src/core/geometry/qgsgeometrycollection.cpp

```diff
@@ -75,6 +82,25 @@ void QgsGeometryCollection::clear()
   clearCache(); //set bounding box invalid
 }
 
+QgsGeometryCollection *QgsGeometryCollection::asGridified( double hSpacing, double vSpacing, double dSpacing, double mSpacing,
+    double tolerance, SegmentationToleranceType toleranceType ) const
+{
+  QgsGeometryCollection *result = nullptr;
+
+  for ( QgsAbstractGeometry *geom : mGeometries )
+  {
+    QgsAbstractGeometry *gridified = geom->asGridified( hSpacing, vSpacing, dSpacing, mSpacing, tolerance, toleranceType );
+    if ( gridified )
+    {
+      if ( !result )
+        result = newSameGeometry();
+      result->mGeometries.append( gridified );
```

Comment:
`result->addGeometry( gridified )`

Reply:
This makes an append with some checks which we already know the outcome of.